src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java

```diff
@@ -134,33 +143,76 @@ public void closeTool() {
 
 
     /**
-     * Goes through the cigar string of the read and create new reads for each consecutive non-N elements (while hard clipping the rest of the read).
-     * For example: for a read with cigar '1H2M2D1M2N1M2I1N1M2S' 3 new reads will be created with cigar strings: 1H2M2D1M, 1M2I and 1M2S
+     * Goes through the cigar string of the read and create new reads for each consecutive non-N elements (while soft clipping the rest of the read) that are supplemental to eachother.
+     * For example: for a read with cigar '1H2M2D1M2N1M2I1N1M2S' 3 new reads will be created with cigar strings: 1H2M2D1M6S, 1H3S1M2I2S and 1H6S1M1S
+     * If the read has an MC tag it will be adjusted according to the clipping of that mate based on its cigar
      *
-     * @param read     the read to split
+     * @param read     the read to split (can be null)
+     * @param emitReads   a parameter used to mock behavior for repairing mate cigar string information
      */
-    public static GATKRead splitNCigarRead(final GATKRead read, OverhangFixingManager manager) {
+    public static GATKRead splitNCigarRead(final GATKRead read, OverhangFixingManager manager, boolean emitReads, SAMFileHeader header, boolean secondaryAlignments) {
         final int numCigarElements = read.numCigarElements();
+        List<GATKRead> splitReads = new ArrayList<>(2);
+        boolean sectionHasMatch = false;
+
+        // Run the tool on dummy mate read to determine what the mate cigar will be upon completion, if manager has a prediction then dont repair
+        if (emitReads && read.hasAttribute("MC")) {
+            read.setAttribute("MC", splitNCigarRead(ArtificialReadUtils.createArtificialRead(header ,TextCigarCodec.decode(read.getAttributeAsString("MC"))), manager, false, header, secondaryAlignments).getCigar().toString());
+        }
+        manager.setPredictedMateInformation(read);
+
+        // If it is a secondary alignment, repair its mate-information (assuming its mate was primary) and pass
+        if ( !secondaryAlignments &&  read.isSecondaryAlignment()){
+            manager.addReadGroup(Collections.singletonList(read));
+            return read;
+        }
 
         int firstCigarIndex = 0;
         for ( int i = 0; i < numCigarElements; i++ ) {
             final CigarElement cigarElement = read.getCigar().getCigarElement(i);
```

Comment:
Instead of using `read.getCigar().getCigarElement(i)` it's better to use `read.getCigarElement(i).  `getCigar` performs a copy of the entire cigar every time which is expensive.  We added the direct getCigarElement recently to avoid that.